File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/CinemaService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.CinemaDTO;
import com.cinema_website.backend.mapper.CinemaMapper;
import com.cinema_website.backend.repository.CinemaRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;


@SuppressWarnings("all")
@Service
public class CinemaService {

    private final CinemaRepository cinemaRepository;
    private final CinemaMapper cinemaMapper;

    @Autowired
    public CinemaService(CinemaRepository cinemaRepository, CinemaMapper cinemaMapper) {
        this.cinemaRepository = cinemaRepository;
        this.cinemaMapper = cinemaMapper;
    }

    // GET
    public CinemaDTO getCinema(int cinemaId) {
        return cinemaRepository.getCinemaById(cinemaId);
    }

    public List<CinemaDTO> getAllCinemas() {
        return cinemaRepository.getAllCinemas();
    }

    // INSERT
    public boolean addCinema(CinemaDTO cinema) {
        return cinemaRepository.addCinema(cinema);
    }

    // UPDATE
    public boolean updateCinema(int cinemaId, CinemaDTO updatedCinema) {
        return cinemaRepository.updateCinemaById(cinemaId, updatedCinema);
    }

    // DELETE
    public boolean deleteCinema(int cinemaId) {
        return cinemaRepository.deleteCinemaById(cinemaId);
    }

}

File: Projects/Project_Cinema Website/frontend/src/services/CinemaService.js
import axios from "axios";

const baseURL = 'http://localhost:8080/api/cinemas'; // Replace with your actual API endpoint

class CinemaService {

    // GET
    static async getCinemaById(cinemaId) {
        const response = await axios.get(`${baseURL}/getCinema/${cinemaId}`);
        return response.data;
    }

    static async getAllCinemas() {
        const response = await axios.get(`${baseURL}/getAllCinemas`);
        return response.data;
    }

    // POST
    static async addCinema(newCinema) {
        const response = await axios.post(`${baseURL}/addCinema`, newCinema);
        return response.data;
    }

    // PUT (Update)
    static async updateCinemaById(cinemaId, updatedCinema) {
        const response = await axios.put(`${baseURL}/updateCinema/${cinemaId}`, updatedCinema);
        return response.data;
    }

    // DELETE
    static async deleteCinemaById(cinemaId) {
        const response = await axios.delete(`${baseURL}/deleteCinema/${cinemaId}`);
        return response.data;
    }
}

export default CinemaService;
